app/src/main/java/com/keylesspalace/tusky/MainActivity.kt

```diff
@@ -0,0 +1,606 @@
+                val accountIdString = intent.getStringExtra(ShortcutManagerCompat.EXTRA_SHORTCUT_ID)
+                if (accountIdString != null) {
+                    accountId = accountIdString.toLong()
+                }
+            }
+            val accountRequested = accountId != -1L
+            if (accountRequested) {
+                val account = accountManager.activeAccount
+                if (account == null || accountId != account.id) {
+                    accountManager.setActiveAccount(accountId)
+                }
+            }
+            if (canHandleMimeType(intent.type)) {
+                // Sharing to Tusky from an external app
+                if (accountRequested) {
+                    // The correct account is already active
+                    forwardShare(intent)
+                } else {
+                    // No account was provided, show the chooser
+                    showAccountChooserDialog(getString(R.string.action_share_as), true, object : AccountSelectionListener {
+                        override fun onAccountSelected(account: AccountEntity) {
+                            val requestedId = account.id
+                            val activeAccount = accountManager.activeAccount
+                            if (activeAccount != null && requestedId == activeAccount.id) {
+                                // The correct account is already active
+                                forwardShare(intent)
+                            } else {
+                                // A different account was requested, restart the activity
+                                intent.putExtra(NotificationHelper.ACCOUNT_ID, requestedId)
+                                changeAccount(requestedId, intent)
+                            }
+                        }
+                    })
+                }
+            } else if (accountRequested) {
+                // user clicked a notification, show notification tab and switch user if necessary
+                showNotificationTab = true
+            }
+        }
+        setContentView(R.layout.activity_main)
+        composeButton.setOnClickListener {
+            val composeIntent = Intent(applicationContext, ComposeActivity::class.java)
+            startActivity(composeIntent)
+        }
+        setupDrawer(savedInstanceState)
+
+        /* Fetch user info while we're doing other things. This has to be done after setting up the
+         * drawer, though, because its callback touches the header in the drawer. */
+        fetchUserInfo()
+
+        setupTabs(showNotificationTab)
+
+        val pageMargin = resources.getDimensionPixelSize(R.dimen.tab_page_margin)
+        viewPager.setPageTransformer(MarginPageTransformer(pageMargin))
+        tabLayout.addOnTabSelectedListener(object : OnTabSelectedListener {
+            override fun onTabSelected(tab: TabLayout.Tab) {
+                if (tab.position == notificationTabPosition) {
```

Comment:
it no longer does `viewPager.setCurrentItem(tab.getPosition());`, is this intended?

Reply:
yes, its not necessary, its handled by the `TabLayoutMediator`